jscomp/main/bsb_main.ml

```diff
@@ -91,6 +91,37 @@ let bsb_main_flags : (string * spec * string) array =
 
 let (//) = Ext_path.combine
 
+let get_version version_string =
+  let components = Str.split (Str.regexp "\\.") version_string in
+  match components with
+  | (major :: minor :: _) -> Some (major ^ "." ^ minor)  
+  | _ -> None
+
+let get_dune_version () =
+  try
+    let prog = Literals.dune ^ " --version" in
+    let inp = Unix.open_process_in prog in
+    let line = input_line inp in
+    close_in inp;
+    get_version line
+  with
+  | Unix.Unix_error(_, _, _) -> None
+
+let make_dune_project_if_does_not_exist proj_dir =
+  let dune_project = proj_dir // Literals.dune_project in
+  if Sys.file_exists dune_project then ()
+  else
+    let dune_version = get_dune_version () in
```

Comment:
let's not shell out to dune.

It should be safe to output `(lang dune 2.8)` (even 2.7 should be enough), as that's the lower bound for the features that we need.

Reply:
Just added a commit removing the shelling out and hardcoding the version to 2.8.